### DISCUSSION TITLE
Vercel deployment, bad performances

### DISCUSSION THREAD

**Discussion by lefuncq** [vercel/next.js#48136] [Help]
### Summary

Hi,

My name is Sacha. I'm the lead front-end senior at Trustblock, and I'm writing this post because, despite my endless efforts to improve performances on our deployments, I'm not getting any convincing results.
Our stack fully complies with Vercel, using serverless functions as our back-end and NextJS as our front-end framework.
The problem mainly occurs on pages loaded using "getServerSideProps". Those pages on our deployment can take up to 7 seconds to be served.
Nonetheless:
- Dev server is fast
- Our serverless functions and database are deployed in the same world region.
- Vercel runtime logs show that cold start duration is never above 300ms.
- Database calls take no longer than 200ms.
- I even tried simplifying our pages to the state where the serverless function, called within getServerSideProps, was straightly reading a single source from our database and sending it back without anything else on the way, and where the page was solely displaying the returned response without any layout or style. I did that because I suspected those long loadings might have been caused by imports within function files and/or page components. And even with that, the outcome was the same.

I've added logs with time to understand better what could cause those long loadings, and my investigation allowed me to discover that it was neither my functions nor my front-end code that were taking long to be resolved but the in-between calls done by Vercel. For example, when a request to get a page is sent, there are at least another two seconds before it even starts getting inside serverSideProps to send the database request from there. As mentioned above, I tried removing anything to test things out in a cleared environment, and the result was the same.

To prevent long loadings every time, we've added several cache headers. Still, every time we release a new version, those caches, expectedly, are invalidated, so loading nightmares resurface, negatively and drastically impacting our business.

The reason we picked Vercel is for its promised performances. Right now, we feel let down cause even from my eight years of seniority in front-end and endless tries and tests I've been conducting for days, I'm unable to come up with a solution that meets even the most standard web conventions in terms of loading times.

So today, I'm asking you for help, and I hope to get it because otherwise, I don't see another way to fix that than to switch to another host, which is the least preferred solution since I love your product so much, for all the fantastic features it provides us with.

<img width="1394" alt="vercel-logs-screenshot" src="https://user-images.githubusercontent.com/15050175/230725639-373fef4b-24f8-4132-846e-0d1bd2c64766.png">

**Comment by Tomekmularczyk** [comment]
@lefuncq are you using pro version of Vercel?
